### PR TITLE
MM-20519 : Migrate 'components/emoji_picker/emoji_picker_overlay.jsx' module to TypeScript

### PR DESCRIPTION
**Pull request by codewithshariq**
#### Summary

This PR migrates 'components/emoji_picker/emoji_picker_overlay.jsx' module and associated tests to TypeScript

#### Ticket Link

Fixes [https://github.com/mattermost/mattermost-server/issues/21155](https://github.com/mattermost/mattermost-server/issues/21155)
JIRA [https://mattermost.atlassian.net/browse/MM-20519](https://mattermost.atlassian.net/browse/MM-20519)

#### Release Note

```release-note

```

### CODE REVIEW

File: components/emoji_picker/emoji_picker_overlay.tsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
 import {Constants} from 'utils/constants';
 
+import {Emoji} from '@mattermost/types/emojis';
+
 import EmojiPickerTabs from './emoji_picker_tabs.jsx';
 
-export default class EmojiPickerOverlay extends React.PureComponent {
+type Props = {
+    show: boolean;
+    container?: () => ReactNode | ReactNode;
```
Comment (M-ZubairAhmed) [inline components/emoji_picker/emoji_picker_overlay.tsx:17]:
```suggestion
    container?: () => ReactNode;
```

Comment (codewithshariq) [reply components/emoji_picker/emoji_picker_overlay.tsx:17]: Hi, @M-ZubairAhmed I referred to React Bootstrap docs before defining types for target and container props. According to React Bootstrap docs, the target and container props can receive DOM nodes and function which returns DOM nodes. What are your thoughts about it?

https://react-bootstrap.github.io/components/overlays/#overlay-props:~:text=API-,%23,-Overlay

Comment (M-ZubairAhmed) [reply components/emoji_picker/emoji_picker_overlay.tsx:17]: i see it goes all the way back to here actually https://github.com/react-restart/ui/blob/de7ce33ea3af51d77e589bf51f41ca071b3c0a6f/src/useWaitForDOMRef.ts 
But the correct type seems to be fine.

**Review by M-ZubairAhmed — APPROVED** [review]
🆗  It looks good to me. Thank you

**Comment by codewithshariq** [comment]
> 🆗 It looks good to me. Thank you

Thank you for the review.

**Review by michelengelen — COMMENTED** [review]
I am wondering why the `hadleEmojiClose` (aka `onEmojiClose`) handlers got removed here? Where they not needed?

**Comment by codewithshariq** [comment]
> I am wondering why the `hadleEmojiClose` (aka `onEmojiClose`) handlers got removed here? Were they not needed?

Yes, those were not needed. In the main component that prop was not even used.

> I am wondering why the `hadleEmojiClose` (aka `onEmojiClose`) handlers got removed here? Were they not needed?

Yes, those were not needed. In the main component that prop was not even used.

**Review by michelengelen — APPROVED** [review]

**Pull request merged**